source/JsonObject.cpp

```diff
@@ -16,26 +16,24 @@ namespace Aws
     {
         JsonObject::JsonObject() : m_wasParseSuccessful(true) { m_value = nullptr; }
 
-        JsonObject::JsonObject(cJSON *value)
-            : m_value(cJSON_Duplicate(value, 1 /* recurse */)), m_wasParseSuccessful(true)
+        JsonObject::JsonObject(aws_json_value *value)
+            : m_value(aws_json_value_duplicate(value)), m_wasParseSuccessful(true)
         {
         }
 
         JsonObject::JsonObject(const String &value) : m_wasParseSuccessful(true)
         {
-            const char *return_parse_end;
-            m_value = cJSON_ParseWithLengthOpts(value.c_str(), value.length(), &return_parse_end, 0);
+            m_value = aws_json_value_new_from_string(ApiAllocator(), aws_byte_cursor_from_c_str(value.c_str()));
 
-            if (m_value == nullptr || cJSON_IsInvalid(m_value) == 1)
+            if (m_value == nullptr || aws_json_value_is_null(m_value) == true)
             {
                 m_wasParseSuccessful = false;
-                m_errorMessage = "Failed to parse JSON at: ";
-                m_errorMessage += return_parse_end;
+                m_errorMessage = "Failed to parse JSON: " + value;
             }
         }
 
         JsonObject::JsonObject(const JsonObject &value)
-            : m_value(cJSON_Duplicate(value.m_value, 1 /*recurse*/)), m_wasParseSuccessful(value.m_wasParseSuccessful),
+            : m_value(aws_json_value_duplicate(value.m_value)), m_wasParseSuccessful(value.m_wasParseSuccessful),
```

Comment:
`aws_json_value_duplicate()` may return NULL, even if something valid was passed in (at least in theory, not sure if possible in reality)

in which case m_wasParseSuccessful should not match value.m_parseWasSuccessful, and m_errorMessage should not match m_errorMessage

might need to move the logic down into the actual function body